scripts/release-notes.py

```diff
@@ -0,0 +1,487 @@
+        pr = lines[0].split(' ', 4)[3]
+        allprs.add(pr)
+        title = lines[2]
+        ncommits, otherauthors = collect_commits(pr, title, list(commit.parents), author[0])
+        individual_authors.add(author[0])
+        individual_authors.update(otherauthors)
+
+        stats = commit.stats.total
+        item = {
+            'title': title,
+            'pr': pr,
+            'sha': commit.hexsha[:shamin],
+            'ncommits': ncommits,
+            'otherauthors': otherauthors,
+            'insertions': stats['insertions'],
+            'deletions': stats['deletions'],
+            'files': stats['files'],
+            'lines': stats['lines'],
+            }
+        history = per_author_history.get(author, [])
+        history.append(item)
+        per_author_history[author] = history
+
+    if len(commit.parents) == 0:
+        break
+    commit = commit.parents[0]
+
+allauthors = list(per_author_history.keys())
+allauthors.sort(key=lambda x:x[0].lower())
+
+ext_contributors = individual_authors - crdb_folk
+firsttime_contributors = []
+for a in individual_authors:
+    # Find all aliases known for this person
+    aliases = [a]
+    for alias, name in author_aliases.items():
+        if name == a:
+            aliases.append(alias)
+    # Collect the history for every alias
+    hist = b''
+    for al in aliases:
+        spin()
+        c = subprocess.run(["git", "log", "--author=%s" % al, options.from_commit, '-n', '1'], stdout=subprocess.PIPE, check=True)
+        hist += c.stdout
+    if len(hist) == 0:
+        # No commit from that author older than the first commit
+        # selected, so that's a first-time author.
+        firsttime_contributors.append(a)
+
+print("\b\n", file=sys.stderr)
+sys.stderr.flush()
+
+### Presentation of results ###
+
+## Print the release notes.
+
+# Start with known sections.
+
+print("---")
+print("title: What&#39;s New in ", end='')
+sys.stdout.flush()
+os.system('git describe --tags ' + options.until_commit)
+print("toc: false")
+print("summary: Additions and changes in CockroachDB version ", end='')
+sys.stdout.flush()
+os.system('git describe --tags ' + options.until_commit)
+print("   since version ", end='')
+sys.stdout.flush()
+os.system('git describe --tags ' + options.from_commit)
+print("---")
+print()
+today = datetime.datetime.today()
+print("## %s %d, %d" % (today.strftime("%B"), today.day, today.year))
+print()
+
+print("This release includes %d merged PRs by %s author%s." %
+      (len(allprs),
+       len(individual_authors), (len(individual_authors) != 1 and "s" or ""),
+      ))
+
+ext_contributors = individual_authors - crdb_folk
+if len(ext_contributors) > 0:
+    ext_contributors = sorted(ext_contributors)
+    print("We would like to thank the following contributors from the CockroachDB community:")
+    print()
+    for a in ext_contributors:
+        print("-", a)
+    print()
+# Note: CRDB folk can be first-time contributors too, so
```

Comment:
It's confusing to see first-time contributors who don't appear in the preceding list (as in [this week's release)](https://www.cockroachlabs.com/docs/releases/v2.0-alpha.20180129.html)). I think we should either exclude CRDB employees from the first-timers list (my first choice) or say something about who these first-time non-community contributors are.

Reply:
Following up in #22206.